### PR TITLE
Pandas numeric columns causing .lower errors. Issue #43

### PR DESCRIPTION
**Pull request by neilsmurphy**
In 2017 the numeric check was dropped from the code. However, the function (colsnumeric) for checking if the dataframe headers were numeric and not string was still in the code. But there was no use of this function any longer. 

Consequently, the current workflow sees the numeric headers not being properly identified and heading into the section where the text headers get sorted and identified. The numeric headers hits a x.lower() designed for strings and an error is thrown, hence this issue.

I have reinstated now the use of the `colsnumeric`.  I have added back in similar fashion (but not exactly the same) code that will set the `self._colmapping` dict which gets used when iterating through the pandas lines to create the backtrader lines.

The order of the columns is obviously important since there is no text to determine the lines. I have checked the [historical codes](https://github.com/mementum/backtrader/commit/05051890efe527ee22919d354038b4dfc1ffe7ca#diff-367c56de63ec5ee081bbbee8f66b1718aecbd507187b4fa39a44398dcb7ad9c5) as well as the [docs](https://www.backtrader.com/docu/pandas-datafeed/pandas-datafeed/) and set the columns accordingly:

{'datetime': None, 'open': 0, 'high': 1, 'low': 2, 'close': 3, 'volume': 4, 'openinterest': 5}

`datetime` being index of the dataframe. 

The `self.colsnumeric` bool qualifier is carried forward and used in the Transform names section in `start()`. This will ensure that numeric headers skip the `.lower()` error and use the second conditional in

```
# Transform names (valid for .ix) into indices (good for .iloc)
        if self.p.nocase and not self.colsnumeric:
            colnames = [x.lower() for x in self.p.dataname.columns.values]
        else:
            colnames = [x for x in self.p.dataname]
```

I will create some test cases but this is working so far. I have run this with numeric and text headers.

### CODE REVIEW

**Comment by neilsmurphy** [comment]
Added test file. (tests/test_data_pandas.py)

**Comment by neilsmurphy** [comment]
The modification is working for regular pandas feed, but not for extended feeds. I'm losing the name in the mapping to the additional lines. Will review.

**Comment by neilsmurphy** [comment]
Everything seems working now with the latest commit. Still unsure how to test in Travis with the extra Pandas module. Travis seems to reject that.

**Comment by neilsmurphy** [comment]
https://towardsdatascience.com/step-by-step-tutorial-to-build-a-minimal-ci-cd-pipeline-for-your-python-project-using-travis-ci-c074e42f0c65

**Comment by neilsmurphy** [comment]
@vladisld Seems all I had to do was modify the travis.yml file to include pandas in the build. All passed. Since I wrote I'll let you authorize.  Thanks.

File: backtrader/feeds/pandafeed.py [hunk 1]
```diff
@@ -28,7 +27,7 @@
 
 
 class PandasDirectData(feed.DataBase):
```
[the remark this answers is not in the document]
Comment (neilsmurphy) [reply backtrader/feeds/pandafeed.py]: Yes that makes sense. Done.

**Comment by neilsmurphy** [comment]
@vladisld Are we good to go with this one?

**Review by vladisld — APPROVED** [review]

**Comment by neilsmurphy** [comment]
@vladisld I adjusted the travis file for the conflict. Are we still good to go?

**Review by vladisld — APPROVED** [review]

**Pull request merged**